KIPs/kip-37.md

```diff
@@ -0,0 +1,908 @@
+
+New functionality is possible with this design such as transferring multiple token types at once, saving on transaction costs. Trading (escrow / atomic swaps) of multiple tokens can be built on top of this standard and it removes the need to “approve” individual token contracts separately. It is also easy to describe and mix multiple fungible or non-fungible token types in a single contract.
+
+## Specification
+
+This document is heavily derived from [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155) written by Witek Radomski, Andrew Cooke, Philippe Castonguay, James Therien, Eric Binet, and Ronan Sandford.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Smart contracts implementing the KIP-37 standard MUST implement all of the functions in the KIP37 interface.
+
+Smart contracts implementing the KIP-37 standard MUST implement the KIP-13 supportsInterface function and MUST return the constant value true if 0xce086315 is passed through the interfaceID argument.
+
+```solidity
+pragma solidity 0.5.6;
+
+/**
+    @title KIP-37 Multi Token Standard
+    @dev See https://kips.klaytn.com/KIPs/kip-37
+    Note: The KIP-13 identifier for this interface is 0xce086315.
+ */
+interface KIP37 /* is KIP13 */ {
+    /**
+        @dev Either `TransferSingle` or `TransferBatch` MUST emit when tokens are transferred, including zero value transfers as well as minting or burning (see "Safe Transfer Rules" section of the standard).
+        The `_operator` argument MUST be the address of an account/contract that is approved to make the transfer (SHOULD be msg.sender).
+        The `_from` argument MUST be the address of the holder whose balance is decreased.
+        The `_to` argument MUST be the address of the recipient whose balance is increased.
+        The `_id` argument MUST be the token type being transferred.
+        The `_value` argument MUST be the number of tokens the holder balance is decreased by and match what the recipient balance is increased by.
+        When minting/creating tokens, the `_from` argument MUST be set to `0x0` (i.e. zero address).
+        When burning/destroying tokens, the `_to` argument MUST be set to `0x0` (i.e. zero address).
+    */
+    event TransferSingle(address indexed _operator, address indexed _from, address indexed _to, uint256 _id, uint256 _value);
```

Comment:
@kaikookim Could you introduce an example of events for mint and create?
What does it look like when a token is created?
What does it look like when a token is minted?

Reply:
I think TransferSingle event and  URI event are enough.
When new token is minted(including creation with initial supply), TransferSingle event should be occured with from address "0x00...0000". Also, the event contains the token ID, so the event covers minting and creating.

Reply:
@kaikookim In that case, can we distinguish it is a creation of a new token type or a mint when a TransferSingle event occurs?

Reply:
I thought that we need to distinguish creation and mint, but Eric and I concluded that a minting event is enough.